src/raw/aws/fargate_fetcher.go

```diff
@@ -102,15 +101,20 @@ func (e *FargateFetcher) getFargateContainerMetrics(stats *FargateStats) error {
 	log.Debug("fargate task stats response from endpoint %s: %s", endpoint, string(response))
 	e.latestFetch = time.Now()
 
+	var stats FargateStats
 	err = json.Unmarshal(response, &stats)
 	if err != nil {
-		return fmt.Errorf("error unmarshalling ECS container: %v", err)
+		return nil, fmt.Errorf("error unmarshalling ECS container: %v", err)
 	}
 
 	now := time.Now()
-	for id := range *stats {
-		(*stats)[id].time = now
+	for i := range stats {
+		if stats[i] == nil {
```

Comment:
FargateStats is
```
type FargateStats map[string]*timedDockerStats
```

Therefore, if I am not wrong, we could also do something like:
```
 for _,k := range stats
    if k == nil{
         k.time....
     }
```

Reply:
Solved in [#9228da6](https://github.com/newrelic/nri-docker/pull/135/commits/9228da6eca7c0622544d78f7fac0584874936e09)